packages/video_player/video_player_avfoundation/darwin/Classes/FVPVideoPlayerPlugin.m

```diff
@@ -828,4 +919,70 @@ - (void)setMixWithOthers:(BOOL)mixWithOthers
 #endif
 }
 
+- (nullable NSNumber *)isPictureInPictureSupported:
+    (FlutterError *_Nullable __autoreleasing *_Nonnull)error {
+  if (@available(macOS 10.15, *)) {
+#if TARGET_OS_OSX
+    return @([AVPictureInPictureController isPictureInPictureSupported]);
+#else
+    return @([AVPictureInPictureController isPictureInPictureSupported] &&
+             [self doesInfoPlistSupportPictureInPicture]);
+#endif
+  } else {
+    return FALSE;
+  }
+}
+
+- (void)setAutomaticallyStartsPictureInPicture:
+            (FVPAutomaticallyStartsPictureInPictureMessage *)input
+                                         error:(FlutterError **)error {
+  FVPVideoPlayer *player = self.playersByTextureId[@(input.textureId)];
+  [player setAutomaticallyStartsPictureInPicture:
+              input.enableStartPictureInPictureAutomaticallyFromInline];
+}
+
+- (void)setPictureInPictureOverlaySettings:(FVPSetPictureInPictureOverlaySettingsMessage *)input
+                                     error:(FlutterError **)error {
+  FVPVideoPlayer *player = self.playersByTextureId[@(input.textureId)];
+  [player
+      setPictureInPictureOverlaySettings:CGRectMake(input.settings.left, input.settings.top,
+                                                    input.settings.width, input.settings.height)];
+}
+
+- (BOOL)doesInfoPlistSupportPictureInPicture {
+  NSArray *backgroundModes = [NSBundle.mainBundle objectForInfoDictionaryKey:@"UIBackgroundModes"];
```

Comment:
It's a type violation to store a non-NSArray-object in an `NSArray*` variable, so if you don't know the type (as indicated by the check below) this should be an `id`.